packages/central-server/__tests__/sync/snapshotOutgoingChanges.test.js

```diff
@@ -590,4 +590,113 @@ describe('snapshotOutgoingChanges', () => {
+        createdAtSyncTick: 1,
+      });
+      await PatientFacility.create({
+        patientId: patient2.id,
+        facilityId: facility.id,
+        lastInteractedTime: new Date('2024-01-01T12:00:00Z'),
+        createdAtSyncTick: 2,
+      });
+      await PatientFacility.create({
+        patientId: patient3.id,
+        facilityId: facility.id,
+        lastInteractedTime: new Date('2024-01-01T14:00:00Z'),
+        createdAtSyncTick: 3,
+      });
+
+      const startTime = new Date();
+      syncSession = await SyncSession.create({
+        startTime,
+        lastConnectionTime: startTime,
+      });
+      await createSnapshotTable(ctx.store.sequelize, syncSession.id);
+    });
+
+    it('limits patients based on lastInteractedThreshold when creating marked for sync table', async () => {
+      const lastInteractedThreshold = 2;
+      
+      const fullSyncPatientsTable = await createMarkedForSyncPatientsTable(
+        ctx.store.sequelize,
+        syncSession.id,
+        true,
+        [facility.id],
+        0,
+        lastInteractedThreshold,
+      );
+ // Query the created table to verify only 2 patients are included
+      const [rows] = await ctx.store.sequelize.query(
+        `SELECT patient_id FROM ${fullSyncPatientsTable} ORDER BY patient_id`
+      );
+
+      expect(rows.length).toBe(2);
+      
+      // Should include the patients with the most recent last_interacted_time
+      const patientIds = rows.map(row => row.patient_id).sort();
+      expect(patientIds).toEqual([patient2.id, patient3.id].sort());
+    });
+
+    it('includes all patients when lastInteractedThreshold is not provided', async () => {
+      const fullSyncPatientsTable = await createMarkedForSyncPatientsTable(
+        ctx.store.sequelize,
+        syncSession.id,
+        true,
+        [facility.id],
+        0,
+        null, // no threshold
+      );
+
+      const [rows] = await ctx.store.sequelize.query(
+        `SELECT patient_id FROM ${fullSyncPatientsTable} ORDER BY patient_id`
+      );
+
+      expect(rows.length).toBe(3);
+      
+      const patientIds = rows.map(row => row.patient_id).sort();
+      expect(patientIds).toEqual([patient1.id, patient2.id, patient3.id].sort());
+    });
+
+    it('works with incremental sync patients table', async () => {
+      const lastInteractedThreshold = 1;
+      
+      const incrementalSyncPatientsTable = await createMarkedForSyncPatientsTable(
+        ctx.store.sequelize,
+        syncSession.id,
+        false,
+        [facility.id],
+        4,
+        lastInteractedThreshold,
+      );
+
+      const [rows] = await ctx.store.sequelize.query(
+        `SELECT patient_id FROM ${incrementalSyncPatientsTable} ORDER BY patient_id`
+      );
+
+      expect(rows.length).toBe(1);
+      
+      // Should include the patient with the most recent last_interacted_time
+      expect(rows[0].patient_id).toBe(patient3.id);
+    });
+  });
```

Comment:
I'd say we need some focused unit tests on full patient sync detection working with the new mechanism:
- It does a full patient sync for patients that are newly marked for sync
- It doesn't do a full patient sync for patients that aren't newly marked for sync, even if they have had a recent interaction (thereby bumping the `updated_at_sync_tick`

There's partial testing of that in here, but with a `since` of `0` in the first case and `4` in the second, we really are focusing on the threshold side of things rather than the full pull vs. incremental pull side of things - which is totally appropriate for this describe block!